packages/block-library/src/column/edit.native.js

```diff
@@ -97,21 +136,36 @@ export default compose( [
 			getBlockCount,
 			getBlockRootClientId,
 			getSelectedBlockClientId,
+			getBlocks,
+			getBlockOrder,
+			getBlock,
 		} = select( 'core/block-editor' );
 
 		const selectedBlockClientId = getSelectedBlockClientId();
 		const isSelected = selectedBlockClientId === clientId;
 
 		const parentId = getBlockRootClientId( clientId );
 		const hasChildren = !! getBlockCount( clientId );
-
 		const isParentSelected =
 			selectedBlockClientId && selectedBlockClientId === parentId;
 
+		const block = getBlock( selectedBlockClientId );
+		const blockOrder = getBlockOrder( parentId );
+
+		const selectedColumnIndex = blockOrder.indexOf( clientId );
+		const columnCount = getBlockCount( parentId );
+		const columnWidths = getBlocks( parentId ).map( ( column ) =>
+			getEffectiveColumnWidth( column, columnCount )
+		);
+		const columnsPreview = { columnWidths, selectedColumnIndex };
+
 		return {
 			hasChildren,
 			isParentSelected,
 			isSelected,
+			columnCount,
+			columnsPreview,
+			block,
```

Comment:
I see that you need the `columnCount` and `block` only for determine the `value`, WDYT about something like:
```suggestion
			columnsPreview,
			value: getEffectiveColumnWidth( block, columnCount ),
```